server/src/Action/HelloAction.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace App\Action;
+
+use Psr\Container\ContainerInterface;
+use Psr\Log\LoggerInterface;
+use Slim\Http\Request;
+use Slim\Http\Response;
+
+class HelloAction {
+
+	/**
+	 * @var ContainerInterface
+	 */
+	private $container;
+
+	/**
+	 * @var LoggerInterface
+	 */
+	private $logger;
+
+	/**
+	 * HelloAction constructor.
+	 * @param ContainerInterface $container
+	 */
+	public function __construct( ContainerInterface $container ) {
+		$this->container = $container;
+		$this->logger = $this->container->get( 'logger' );
+	}
```

Comment:
You should inject the dependency, not the container.

i.e. inject a `$logger`

Or is this the only way it can be done? Perhaps the action should be within the container as well? Then the route can just call something like `$container->get('hello_action')->helloAction()` (if necessary)

Reply:
The reason I point this out, is it's a lot easier to mock a LoggerInterface than to mock the entire container in testing.

Reply:
I agree, only needed dependencies should be injected, but since this is a test `Action` and it is not declared in the container, slim itself adds the container as a dependency. I didn't bother changing it.

I will remove the whole action on the next PR when we have an endpoint